crates/lib/kajiya-backend/src/vulkan/image.rs

```diff
@@ -246,11 +246,41 @@ impl Default for ImageViewDesc {
 impl Device {
     pub fn create_image(
         &self,
-        desc: ImageDesc,
+        mut desc: ImageDesc,
         initial_data: Vec<ImageSubResourceData>,
     ) -> Result<Image, BackendError> {
         log::info!("Creating an image: {:?}", desc);
 
+        let limits = self.pdevice.properties.limits;
+
+        let max_dimension = if desc.extent[2] > 1 {
+            limits
+                .max_image_dimension1_d
+                .min(limits.max_image_dimension2_d)
+                .min(limits.max_image_dimension3_d)
+        } else if desc.extent[1] > 1 {
+            limits
+                .max_image_dimension1_d
+                .min(limits.max_image_dimension2_d)
+        } else {
+            limits.max_image_dimension1_d
+        };
+
+        for i in 0..3 {
+            let dimension = desc.extent[i];
+
+            if dimension > max_dimension {
+                log::warn!(
+                    "Dimension {} ({}) exceeds max dimension {}. Adjusting to the max. {:?}",
+                    i,
+                    dimension,
+                    max_dimension,
+                    desc
+                );
+                desc.extent[i] = max_dimension;
+            }
+        }
```

Comment:
Ah, just checked and the problematic volume texture is coming from CSGI because the indirect cascade texture is
```rust
[
    VOLUME_DIMS * TOTAL_SUBRAY_COUNT as u32),
    VOLUME_DIMS,
    VOLUME_DIMS,
]
```
or `[3456, 64, 64]`.

Can this be 1x1x1 as well?

Reply:
Yup, I believe it can!